pkg/controllers/monitoring-stack/controller.go

```diff
@@ -130,20 +131,74 @@ func (r *reconciler) Reconcile(ctx context.Context, req ctrl.Request) (ctrl.Resu
 		return ctrl.Result{}, nil
 	}
 
+	availableCondition := metav1.Condition{
+		Type:               "Available",
+		ObservedGeneration: ms.ObjectMeta.Generation,
+		LastTransitionTime: metav1.Time{Time: time.Now().UTC()},
+	}
+
 	reconcilers := stackComponentReconcilers(ms, r.instanceSelectorKey, r.instanceSelectorValue)
 	for _, reconciler := range reconcilers {
 		err := reconciler(ctx, r.k8sClient, r.scheme)
 		// handle create / update errors that can happen due to a stale cache by
 		// retrying after some time.
 		if errors.IsAlreadyExists(err) || errors.IsConflict(err) {
 			logger.V(3).Info("skipping reconcile error", "err", err)
+			availableCondition.Status = metav1.ConditionFalse
+			availableCondition.Reason = "Failure"
+			availableCondition.Message = fmt.Sprintf("Monitoring stack failed to reconcile: %v", err)
+			err := r.updateStatusCondition(ctx, ms, availableCondition)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
 			return ctrl.Result{RequeueAfter: 2 * time.Second}, nil
 		}
 		if err != nil {
+			availableCondition.Status = metav1.ConditionFalse
+			availableCondition.Reason = "Failure"
+			availableCondition.Message = fmt.Sprintf("Monitoring stack failed to reconcile: %v", err)
+			err := r.updateStatusCondition(ctx, ms, availableCondition)
+			if err != nil {
+				return ctrl.Result{}, err
+			}
 			return ctrl.Result{}, err
 		}
 	}
-	return ctrl.Result{}, nil
+	availableCondition.Status = metav1.ConditionTrue
+	availableCondition.Reason = "Ready"
+	availableCondition.Message = "Monitoring stack is up and running"
```

Comment:
This is misleading ... since the stack can fail for various other reasons. A better message may be "Monitoring stack has been deployed" or something on those lines.

Reply:
What about "Monitoring stack has been reconciled" (or updated)? "Deployed" sounds like one time action to me.

Reply:
I agree `Deployed` does sound like a one time action.  I feel `updated` sounds better than reconciled.